### PR TITLE
[FLINK-34130][core] Mark setBytes and getBytes of Configuration as @Internal

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What is the purpose of the change

[FLINK-34130][core] Mark setBytes and getBytes of Configuration as @Internal

## Brief change log

[FLINK-34130][core] Mark setBytes and getBytes of Configuration as @Internal

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Review by 1996fanrui — COMMENTED** [review]
Hi @Sxnan @RocMarshal , would you mind reviewing this PR in your free time? thanks~

**Review by Sxnan — APPROVED** [review]
@1996fanrui  Thanks for the PR! LGTM!

**Review by RocMarshal — APPROVED** [review]
@1996fanrui Thanks for the PR.
LGTM +1.

**Comment by 1996fanrui** [comment]
@flinkbot run azure

File: pom.xml [hunk 1]
```diff
@@ -2383,6 +2383,9 @@ under the License.
 								<exclude>org.apache.flink.util.function.SerializableFunction</exclude>
 								<exclude>org.apache.flink.util.function.SupplierWithException</exclude>
 								<exclude>org.apache.flink.util.function.ThrowingConsumer</exclude>
+								<!-- Mark these 2 methods to @Internal. Tracked under FLINK-34130 -->
+								<exclude>org.apache.flink.configuration.Configuration#getBytes()</exclude>
```
[the remark this answers is not in the document]
Comment (1996fanrui) [reply pom.xml]: @reswqa Thank you very much for point it out!

[the remark this answers is not in the document]
Comment (1996fanrui) [reply pom.xml]: I try build it with `byte[]` on my Mac. Both of Jdk 8 and jdk 17 are fine.

And the CI seems good to me for now.

**Comment by 1996fanrui** [comment]
Thanks all of you for the review and suggestion! Merging~

**Pull request merged**